### ISSUE TITLE
onediff cause inconsistent results between two runs.

### ISSUE THREAD

**Issue by HydrogenQAQ** [siliconflow/onediff#773]
#### A clear and concise description of what the bug is.

Hello, when I executed the following script twice, the results using pytorch were consistent, but the results using onediff were inconsistent.

#### OS 

diffusers git commit id: 76c645d3a641c879384a

#### OneDiff git commit id

b89dd4980aeab85b8f9

####  OneFlow version info
Run `python -m oneflow --doctor` and paste it here.

```
version: 0.9.1.dev20240124+cu121
git_commit: 6458a12
cmake_build_type: Release
rdma: True
mlir: True
enterprise: False
```

### How To Reproduce
Steps to reproduce the behavior(code or script):

```
import argparse

from diffusers import (
    StableDiffusionPipeline,
)
import torch

parser = argparse.ArgumentParser()
parser.add_argument("--base", type=str, default="runwayml/stable-diffusion-v1-5")

parser.add_argument("--prompt", type=str, default="a boy")
parser.add_argument("--height", type=int, default=512)
parser.add_argument("--width", type=int, default=512)
parser.add_argument("--n_steps", type=int, default=30)
parser.add_argument("--saved_image", type=str, required=False, default="out.png")
parser.add_argument(
    "--use_onediff", type=(lambda x: str(x).lower() in ["true", "1", "yes"]), default=False
)

args = parser.parse_args()

print("build pipeline")
pipe = StableDiffusionPipeline.from_pretrained(args.base, torch_dtype=torch.float16)
pipe.safety_checker = None
pipe.requires_safety_checker = False
pipe.to("cuda")

# onediff
if args.use_onediff:
    from onediff.infer_compiler import oneflow_compile

    pipe.unet = oneflow_compile(pipe.unet)

    pipe.vae.encoder = oneflow_compile(pipe.vae.encoder)
    pipe.vae.decoder = oneflow_compile(pipe.vae.decoder)

    # pipe.controlnet = oneflow_compile(pipe.controlnet)

run_image = pipe(
    args.prompt,
    height=args.height,
    width=args.width,
    num_inference_steps=args.n_steps,
    generator=torch.manual_seed(10)
).images[0]

run_image.save(args.saved_image)
```

The script test.py is as shown above, and then executes using pytorch and using onediff:

```
python test.py --saved_image torch_out1.png
python test.py --saved_image torch_out2.png

python test.py --saved_image onediff_out1.png --use_onediff 1
python test.py --saved_image onediff_out2.png --use_onediff 1
```

Finally compare the results：

```
from PIL import Image

torch_image_1 = Image.open('torch_out1.png')
torch_image_2 = Image.open('torch_out2.png')

onediff_image_1 = Image.open('onediff_out1.png')
onediff_image_2 = Image.open('onediff_out2.png')

for pair in [(torch_image_1, torch_image_2), (onediff_image_1, onediff_image_2)]:
    pixels1 = list(pair[0].getdata())
    pixels2 = list(pair[1].getdata())

    differences = 0
    for i in range(len(pixels1)):
        if pixels1[i] != pixels2[i]:
            differences += 1

    total_pixels = 512 * 512
    percentage_difference = (differences / total_pixels) * 100

    print(f"The percentage difference between the two images is: {percentage_difference}%")
```

the result is:

```
The percentage difference between the two images is: 0.0%
The percentage difference between the two images is: 79.81147766113281%
```

**Comment by strint** [comment]
![image](https://github.com/siliconflow/onediff/assets/3468789/07e5f5b9-cd5b-4945-99df-ad7c8d1099d9)

same seed
- run one process and exit;
- run another process and exit;

got some difference.

Want: need the results to be reproducible at the pixel level.

**Comment by clackhan** [comment]
@HydrogenQAQ you use save/load pipe to get the the results to be reproducible at the pixel level. just like the below code:

```python
import argparse

from diffusers import (
    StableDiffusionPipeline,
)
import torch

parser = argparse.ArgumentParser()
parser.add_argument("--base", type=str, default="/data/hf_models/stable-diffusion-v1-5")

parser.add_argument("--prompt", type=str, default="a boy")
parser.add_argument("--height", type=int, default=512)
parser.add_argument("--width", type=int, default=512)
parser.add_argument("--n_steps", type=int, default=30)
parser.add_argument("--saved_image", type=str, required=False, default="out.png")
parser.add_argument(
    "--use_onediff", type=(lambda x: str(x).lower() in ["true", "1", "yes"]), default=False
)

parser.add_argument(
    "--load_pipe", type=(lambda x: str(x).lower() in ["true", "1", "yes"]), default=False
)

parser.add_argument(
    "--save_pipe", type=(lambda x: str(x).lower() in ["true", "1", "yes"]), default=False
)

args = parser.parse_args()

print("build pipeline")
pipe = StableDiffusionPipeline.from_pretrained(args.base, torch_dtype=torch.float16)
pipe.safety_checker = None
pipe.requires_safety_checker = False
pipe.to("cuda")

# onediff
if args.use_onediff:
    from onediff.infer_compiler import oneflow_compile

    from onediffx import compile_pipe, save_pipe, load_pipe

    pipe = compile_pipe(pipe)

    if args.load_pipe:
        load_pipe(pipe, "sd_15_pipe")

    # pipe.vae.encoder = oneflow_compile(pipe.vae.encoder)
    # pipe.vae.decoder = oneflow_compile(pipe.vae.decoder)

    # pipe.controlnet = oneflow_compile(pipe.controlnet)

run_image = pipe(
    args.prompt,
    height=args.height,
    width=args.width,
    num_inference_steps=args.n_steps,
    generator=torch.manual_seed(10)
).images[0]

run_image.save(args.saved_image)
if args.use_onediff and args.save_pipe:
    save_pipe(pipe, "sd_15_pipe")
```
executes test.py
```
python test.py --saved_image onediff_out1.png --use_onediff 1 --save_pipe 1
python test.py --saved_image onediff_out2.png --use_onediff 1 --load_pipe 1
```
